cli/azd/pkg/alpha/alpha_feature_manager.go

```diff
@@ -77,32 +79,32 @@ func (m *FeatureManager) IsEnabled(featureId FeatureId) bool {
 	// guard from using the alphaFeatureManager from multiple routines. Only the first one will create the cache.
 	m.withSync.Do(m.initConfigCache)
 
+	enabled := false
+
 	// For testing, and in CI, allow enabling alpha features via the environment.
-	envName := fmt.Sprintf("AZD_ALPHA_ENABLE_%s", strings.ToUpper(string(featureId)))
+	envName := fmt.Sprintf("AZD_ALPHA_ENABLE_%s", strings.ReplaceAll(strings.ToUpper(string(featureId)), ".", "_"))
 	if v, has := os.LookupEnv(envName); has {
 		if b, err := strconv.ParseBool(v); err == nil {
-			return b
+			enabled = b
 		} else {
 			log.Printf("could not parse %s as a bool when considering %s", v, envName)
 		}
+	} else if allOn := isEnabled(m.userConfigCache, AllId); allOn {
+		//check if all features is ON
+		enabled = true
+	} else if featureOn := isEnabled(m.userConfigCache, featureId); featureOn {
+		// check if the feature is ON
+		enabled = true
+	} else if val, ok := defaultEnablement[strings.ToLower(string(featureId))]; ok {
+		// check if the feature has been set with a default value internally
+		enabled = val
 	}
 
-	//check if all features is ON
-	if allOn := isEnabled(m.userConfigCache, AllId); allOn {
-		return true
-	}
-
-	// check if the feature is ON
-	if featureOn := isEnabled(m.userConfigCache, featureId); featureOn {
-		return true
-	}
-
-	// check if the feature has been set with a default value internally
-	if val, ok := defaultEnablement[strings.ToLower(string(featureId))]; ok {
-		return val
+	if enabled {
+		tracing.AppendUsageAttribute(fields.AlphaFeaturesKey.StringSlice([]string{string(featureId)}))
```

Comment:
Should we check if the attribute is already present in the trace?  or would that happen automatically?
This method could be called multiple times during a command execution

Reply:
Looks like @weikanglim has a PR #4250 out to track unique values.  I will migrate to that once his PR has been merged.